remoteconfig/condition_evaluator.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2025 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remoteconfig
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"log"
+	"math/big"
+)
+
+type conditionEvaluator struct {
+	evaluationContext map[string]any
+	conditions        []namedCondition
+}
+
+const (
+	maxConditionRecursionDepth = 10
+	randomizationID            = "randomizationID"
+	rootNestingLevel           = 0
+	totalMicroPercentiles      = 100_000_000
+)
+
+const (
+	lessThanOrEqual = "LESS_OR_EQUAL"
+	greaterThan     = "GREATER_THAN"
+	between         = "BETWEEN"
+)
+
+func (ce *conditionEvaluator) evaluateConditions() map[string]bool {
+	evaluatedConditions := make(map[string]bool)
+	for _, condition := range ce.conditions {
+		evaluatedConditions[condition.Name] = ce.evaluateCondition(condition.Condition, rootNestingLevel)
+	}
+	return evaluatedConditions
+}
+
+func (ce *conditionEvaluator) evaluateCondition(condition *oneOfCondition, nestingLevel int) bool {
+	if nestingLevel >= maxConditionRecursionDepth {
+		log.Println("Maximum recursion depth is exceeded.")
```

Comment:
Just curious: are we okay to log these instead of throwing?

Reply:
These log statements have been added as supporting statements to the early `false` returns; more of an implementation detail.

Additionally, wanted it to be consistent with the [Node](https://github.com/firebase/firebase-admin-node/blob/c3090575682fd09957ae675ea6024ddab2d09406/src/remote-config/condition-evaluator-internal.ts#L63) and [Python](
https://github.com/firebase/firebase-admin-python/blob/bde3fb0134b2f84d789cff47b932c07981d6565b/firebase_admin/remote_config.py#L363) SDK implementations.

However, if it is preferred that we throw rather than log; we will make the change 👍🏻